packages/react-client/src/ReactFlightClient.js

```diff
@@ -197,6 +223,55 @@ function createElement(type, key, props): React$Element<any> {
   return element;
 }
 
+type UninitializedBlockPayload<Data> = [
+  mixed,
+  ModuleMetaData | Chunk<ModuleMetaData>,
+  Data | Chunk<Data>,
+];
+
+type Thenable<T> = {
+  then(resolve: (T) => mixed, reject?: (mixed) => mixed): Thenable<any>,
+};
+
+function initializeBlock<Props, Data>(
+  tuple: UninitializedBlockPayload<Data>,
+): BlockComponent<Props, Data> {
+  // Require module first and then data. The ordering matters.
```

Comment:
Can we test correct ordering?

Reply:
Hm. Yea. We could test a side-effect in the require somehow. There's no module function that runs in the tests atm though so needs some infra or hook into the mocks somehow.

Reply:
We also don't have any async tests yet.